### PR TITLE
core: add BLS12_381 interops

### LINKED ISSUE #2921: Groth16
See neo-project/neo#2691.

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Close #2921.

A very initial implementation, TBD:

- [x] Implement scalar multiplication operation for pairings.
- [x] Update StdLib API where possible (RPC client, interops, wrappers, etc.).
- [x] Port the whole batch of C# cryptography tests (~really a lot~ there's not so much tests with meaningful numbers).
- [x] Write some tests for general functionality.

Note about BLS12_381 library choice: I've firstly consider the etherium library (https://github.com/ethereum/go-ethereum/tree/v1.11.4/crypto/bls12381), but it doesn't contain compressed points serialization (only uncompressed one). Thus, current implementation uses https://github.com/ConsenSys/gnark-crypto/blob/master/ecc/ecc.md.

### CODE REVIEW

**Comment by AnnaShaleva** [comment]
Ref. #2949? currentlu we don't have RPC client wrapper for CryptoLib.

File: pkg/core/native/crypto.go [hunk 1]
```diff
@@ -137,6 +176,289 @@ func curveFromStackitem(si stackitem.Item) (elliptic.Curve, error) {
...
+			panic(errors.New("extra bytes after compressed bls12381 G1Affine point"))
+		}
+		res = g1Affine
+	case bls12381.SizeOfG2AffineCompressed:
+		g2Affine := new(bls12381.G2Affine)
+		read, err := g2Affine.SetBytes(buf)
+		if err != nil {
+			panic(fmt.Errorf("failed to decode bls12381 G2Affine point: %w", err))
+		}
+		if read != l {
```
[the remark this answers is not in the document]
Comment (AnnaShaleva) [reply pkg/core/native/crypto.go]: Fixed.

File: pkg/core/native/native_test/cryptolib_test.go [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+	stack, err := c.TestInvokeScript(t, script.Bytes(), c.Signers)
+	require.NoError(t, err)
+	require.Equal(t, 1, stack.Len())
+	itm := stack.Pop().Item()
+	require.Equal(t, stackitem.BooleanT, itm.Type())
+	require.True(t, itm.Value().(bool))
+}
+
+// Data used for this test are generated by C# library.
+func TestCryptolib_TestBls12381Mul_CompatCustom(t *testing.T) {
```
Comment (AnnaShaleva) [inline pkg/core/native/native_test/cryptolib_test.go:185]: @roman-khimov, do we need more tests like that for other operations/points? This test uses exact points represented as hex-encoded strings that were generated randomply. The `expected` result is taken "as is" directly from the C# operation result.

Comment (roman-khimov) [reply pkg/core/native/native_test/cryptolib_test.go:185]: It'd be nice to have _exactly the same_ data in C# tests. This way we'll ensure that those number really are correct and that they won't break in the future.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**